### PR TITLE
VIM-XXXX: Adds Carthage support, updates README

### PR DESCRIPTION
**Pull request by rpassis**
#### Ticket

[VIM-XXXX](https://vimean.atlassian.net/browse/VIM-XXXX)

- *Note: this is required for Vimeo staff only. If not applicable to your PR, use "N/A".*

#### Pull Request Checklist

- [x] Resolved any merge conflicts
- [x] No build errors or warnings are introduced
- [x] New files are written in Swift
- [x] New classes contain license headers
- [x] New classes have Documentation
- [x] New public methods have Documentation

#### Issue Summary

Adds Carthage support to the library

#### Implementation Summary

- Shared framework scheme for tvOS and iOS 
- Updated README

#### How to Test

To test simply run `carthage build --no-skip-current`.

Once completed the `Carthage/Build` folder should exist and inside it you will find the compiled framework file under the applicable platform (iOS and tvOS in this case).

### CODE REVIEW

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          6 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/f3f7f0f7d877ce4ba69cd612754ad10cd07ed714/Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L75">Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L75</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/f3f7f0f7d877ce4ba69cd612754ad10cd07ed714/Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L95">Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L95</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/f3f7f0f7d877ce4ba69cd612754ad10cd07ed714/Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L104">Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L104</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/f3f7f0f7d877ce4ba69cd612754ad10cd07ed714/Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L76">Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L76</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/f3f7f0f7d877ce4ba69cd612754ad10cd07ed714/Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L97">Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L97</a></strong>:   <br />```</p>

<p>```</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/f3f7f0f7d877ce4ba69cd612754ad10cd07ed714/Pods/AFNetworking/AFNetworking/AFURLSessionManager.m#L163">Pods/AFNetworking/AFNetworking/AFURLSessionManager.m#L163</a></strong>: ‘setResumingHandler:’ is only available on iOS 9.0 or newer [-Wunguarded-availability]  <br /><code>
        [self.uploadProgress setResumingHandler:^{
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 114 tests, with 0 failures (0 unexpected) in 1.017 (1.128) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/1047)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by rpassis** [comment]
Hey can you please try without chdir into the project folder? Carthage should be able to handle that itself. I just tried it with a fresh copy of the repo, checked out the Carthage branch and ran the command above - all worked as expected.

Screen Shot 2019-06-04 at 6 25 52 AM

**Comment by rpassis** [comment]
I also created a sample integration project [here](https://github.com/rpassis/VIMNetworkingCarthageSample).

You can clone it and integrate it with `carthage bootstrap --platform ios` and the project should build and run as expected.

**Comment by rpassis** [comment]
Also FWIW I've added a `Cartfile` to list the dependency on AFNetworking, and created an integration sample projects for you to take a look - you can clone and then run `carthage update --platform ios` and it should automatically build `VimeoNetworking` and `AFNetworking`, place both under the `Carthage/Build/iOS` folder AND you should be able to build and run with no issues.

**Comment by nicolelehrer** [comment]
thanks, @rpassis - I did initially do it in the root folder but it couldn't find the project file. Also tried a  fresh clone but encountered same. Though I'm going to retry with your new changes in between sessions and keep you posted 👍thanks!

File: Cartfile [hunk 1]
```diff
@@ -0,0 +1 @@
+github "AFNetworking/AFNetworking" == 3.1.0
```
Comment (nicolelehrer) [inline Cartfile]: not - Should we add a new line here?

**Pull request merged**